ta4j-core/src/main/java/org/ta4j/core/indicators/caching/BaseIndicatorValueCache.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2023 Ta4j Organization & respective
+ * authors (see AUTHORS)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package org.ta4j.core.indicators.caching;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import org.ta4j.core.BarSeries;
+
+import java.time.ZonedDateTime;
+import java.util.Collections;
+import java.util.Map;
+import java.util.function.Function;
+
+public class BaseIndicatorValueCache<T> implements IndicatorValueCache<T> {
+
+    private final Cache<ZonedDateTime, T> cache;
+
+    public BaseIndicatorValueCache(BarSeries series) {
+        this(new IndicatorValueCacheConfig(series.getMaximumBarCount()));
```

Comment:
IIRC when the bar series maximum bar count isn't set it'll default to `Integer.MAX_VALUE`. When this value is passed to Caffeine as the maximumSize it'll implement the cache using a `BoundedLocalCache` (a relatively heavy-weight object) as opposed to an `UnboundedLocalCache` (a light-weight wrapper around ConcurrentHashMap). In practice I don't know how much performance impact it'll have but if we have a lot of indicators active it could be something. 

Adding a check here (or perhaps within the `IndicatorValueCacheConfig` constructor) to identify the `Integer.MAX_VALUE` scenario and then either not set Caffeine's maximumSize or pass -1 (what it uses for unbounded) could be worthwhile.

Reply:
Good point, using the `unboundedLocaleCache` seems to be the better default option here